sdk/servicebus/service-bus/test/stress/stressTestsBase.ts

```diff
@@ -73,42 +88,60 @@ export class SBStressTestsBase {
         "close-info"
       ];
     }
-    if (!this.snapshotOptions.snapshotIntervalInMs) {
-      this.snapshotOptions.snapshotIntervalInMs = 5000;
-    }
+
+    const snapshotIntervalMs = !this.snapshotOptions.snapshotIntervalInMs ? 5000 : this.snapshotOptions.snapshotIntervalInMs;
+
     this.startedAt = new Date();
     this.messagesSent = [];
     this.snapshotTimer = setInterval(
       this.snapshot.bind(this),
-      snapshotOptions.snapshotIntervalInMs
+      snapshotIntervalMs
     );
   }
 
+  /**
+   * Creates a ServiceBusClient using the connection string in the SERVICEBUS_CONNECTION_STRING environment variable.
+   */
+  public createServiceBusClient(): ServiceBusClient {
+    if (!process.env.SERVICEBUS_CONNECTION_STRING) {
+      throw new Error("Failed to create a ServiceBusClient - no connection string defined in the environment");
+    }
+
+    return new ServiceBusClient(process.env.SERVICEBUS_CONNECTION_STRING);
+  }
+
   public async init(
     queueNamePrefix?: string,
     options?: CreateQueueOptions | undefined,
     testOptions?: Record<string, string | number | boolean>
   ) {
-    this.queueName =
-      (!queueNamePrefix ? `queue` : queueNamePrefix) + `-${Math.ceil(Math.random() * 100000)}`;
-    this.messagesReportFileName = `temp/messages-${this.queueName}.json`;
-    if (testOptions) console.log(testOptions);
-
-    defaultClient.commonProperties = {
-      // these will be reported with each event
-      testName: this.snapshotOptions.testName
-    };
+    try {
+      this.queueName =
+        (!queueNamePrefix ? `queue` : queueNamePrefix) + `-${Math.ceil(Math.random() * 100000)}`;
+      if (testOptions) console.log(testOptions);
 
-    defaultClient.trackEvent({
-      name: "start",
-      properties: {
-        ...testOptions,
-        ...options,
-        queueName: this.queueName
-      }
-    });
+      defaultClient.commonProperties = {
+        // these will be reported with each event
+        testName: this.snapshotOptions.testName
+      };
 
-    await this.serviceBusAdministrationClient.createQueue(this.queueName, options);
+      defaultClient.trackEvent({
+        name: "start",
+        properties: {
+          ...testOptions,
+          ...options,
+          queueName: this.queueName
+        }
+      });
+
+      await this.serviceBusAdministrationClient.createQueue(this.queueName, options);
+    } catch (err) {
+      this.trackError("init", err);
+      // TODO: we might want to consider just having a .run(() => { <your-code>}) style so we can avoid the
+      // possibility of _losing_ fatal telemetry because your app exited without flushing.
```

Comment:
Good point!

Reply:
:)

Reply:
The other option (based on how you created stressTestBase) would be to treat it as a template object. That would also be fine by me.